### ISSUE TITLE
Migrate from cactoos some matchers leftover in their test packages

### ISSUE THREAD

**Issue by llorllale** [llorllale/cactoos-matchers#17]
We missed these matchers found only in `cactoos` test packages. We need to bring them over here:

* BehavesAsCollection
* BehavesAsList
* BehavesAsMap
* ClearDeletesAllValues
* PutUpdatesValues
* PutAllUpdatesValues
* RemoveDeletesValues

We'll release `1.0` once this is complete.

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/llorllale/cactoos-matchers/issues/17#issuecomment-392592600))

@llorllale Job [#17](https://github.com/llorllale/cactoos-matchers/issues/17) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by 0crat** [comment]
The job [#17](https://github.com/llorllale/cactoos-matchers/issues/17) assigned to @dfrau32[/z](https://www.0crat.com/u/dfrau32), here is [why](https://www.0crat.com/footprint/C63314D6Z/1861318972); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by dfrau32** [comment]
@0crat resign

**Comment by 0crat** [comment]
> @0crat resign ([here](https://github.com/llorllale/cactoos-matchers/issues/17#issuecomment-405080971))

@dfrau32 The user @dfrau32[/z](https://www.0crat.com/u/dfrau32) resigned from [#17](https://github.com/llorllale/cactoos-matchers/issues/17), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Tasks refusal is discouraged, see [§6](http://www.zerocracy.com/policy.html#6): -15 point(s) just awarded to @dfrau32[/z](https://www.0crat.com/u/dfrau32)

**Comment by llorllale** [comment]
@0crat out because these matchers seem unfocused..

**Comment by 0crat** [comment]
> @0crat out because these matchers seem unfocused.. ([here](https://github.com/llorllale/cactoos-matchers/issues/17#issuecomment-445489533))

@llorllale Job `gh:llorllale/cactoos-matchers#17` is not assigned, can't get performer

**Comment by 0crat** [comment]
> @0crat out because these matchers seem unfocused.. ([here](https://github.com/llorllale/cactoos-matchers/issues/17#issuecomment-445489533))

@llorllale The job [#17](https://github.com/llorllale/cactoos-matchers/issues/17) is now out of scope
